From 32a044166e9c02466660549983589bbec39f5752 Mon Sep 17 00:00:00 2001
From: Mike Mammarella <mdm@chromium.org>
Date: Thu, 24 Sep 2009 17:03:24 +0000
Subject: [PATCH 02/23] Fix expr.c linker error.

This can be seen when compiled with a sufficiently optimizing compiler.

Original review URL: http://codereview.chromium.org/231010
---
 third_party/sqlite/src/src/expr.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/third_party/sqlite/src/src/expr.c b/third_party/sqlite/src/src/expr.c
index c0e9ba6..2699ae1 100644
--- a/third_party/sqlite/src/src/expr.c
+++ b/third_party/sqlite/src/src/expr.c
@@ -781,7 +781,9 @@ static Expr *exprDup(sqlite3 *db, Expr *p, int flags, u8 **pzBuffer){
       }else{
         int nSize = exprStructSize(p);
         memcpy(zAlloc, p, nSize);
-        memset(&zAlloc[nSize], 0, EXPR_FULLSIZE-nSize);
+        if( EXPR_FULLSIZE>nSize ){
+          memset(&zAlloc[nSize], 0, EXPR_FULLSIZE-nSize);
+        }
       }
 
       /* Set the EP_Reduced, EP_TokenOnly, and EP_Static flags appropriately. */
-- 
2.2.1

